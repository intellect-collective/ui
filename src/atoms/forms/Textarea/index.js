import React from 'react';
import changeable from '../_decorators/changeable';
import formContext from '../_decorators/formContext';

const Textarea = (props) => (
    <textarea { ...props } />
);
Textarea.displayName = 'Textarea';

export default formContext(changeable(Textarea));
