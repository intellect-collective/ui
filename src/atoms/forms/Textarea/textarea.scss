@if (null != index($ui-includes, 'form.textarea') and null == index($ui-excludes, 'form.textarea')) {
//	textarea:not(.browser-default) {
//	    @extend .input;
//	}
}