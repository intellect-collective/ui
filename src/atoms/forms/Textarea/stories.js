import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Textarea from './index';
import width from '../../../../.storybook/decorators/width';

storiesOf('Atoms / Forms / Textarea', module)
    .addDecorator(width(200))
    .add('default', () => (<Textarea name="test" placeholder="Test..." onChange={ action('onChange') } />));
