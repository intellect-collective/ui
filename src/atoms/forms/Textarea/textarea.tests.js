import React from 'react';
import Textarea from '.';
import {
    basicChangeableTests
} from '../../../../test/utils';

const validProps = {
    name: 'test',
    value: 'secret',
    onChange: () => {}
};
const expectedProps = {
    name: 'test',
    value: 'secret',
    onChange: expect.any(Function)
};

describe('Textarea', () => {
    basicChangeableTests(Textarea, 'textarea', validProps, expectedProps);
});
