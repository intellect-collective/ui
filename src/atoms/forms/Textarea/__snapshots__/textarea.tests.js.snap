// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea Basic changeable tests Should accept valid props 1`] = `
<FormContext
  name="test"
  onChange={[Function]}
  value="secret"
>
  <FormContextInner
    ctx={null}
    name="test"
    onChange={[Function]}
    value="secret"
  >
    <Changeable
      name="test"
      onChange={[Function]}
      value="secret"
    >
      <Textarea
        name="test"
        onChange={[Function]}
        value="secret"
      >
        <textarea
          name="test"
          onChange={[Function]}
          value="secret"
        />
      </Textarea>
    </Changeable>
  </FormContextInner>
</FormContext>
`;
